src/options.ts

```diff
@@ -120,6 +120,13 @@ export interface Options {
    * Defaults to `true` when calling the APIs per `setup`.
    */
   writeToClipboard?: boolean
+
+  /**
+   * A function to be called internally to advance your fake timers (if applicable)
+   *
+   * @example jest.advanceTimersByTime
+   */
+  advanceTimers?: (delay: number) => void
```

Comment:
I think this should return `Promise<void>`. The resulting `await` on this enables the user to let third-party code run after the timers were advanced and before we continue interacting. Otherwise asynchronous code during the timeout might not be executed until after the next interaction which might result in hard to debug issues.

Reply:
The place where this `advanceTimers` function would be used is already inside of a promise and resolving the promise is already from a `setTimeout`. Can you give an example of some code which would need `advanceTimers` to return a promise?

Reply:
```ts
function wait(t: number) {
    return new Promise<void>(r => setTimeout(r, t))
}
function waitVoidAdvance(t: number, advance: (t: number) => void) {
    return new Promise<void>(resolve => {
        setTimeout(() => resolve(), t)
        advance(t)
    })
}
function waitAwaitAdvance(t: number, advance: (t: number) => Promise<unknown> | unknown) {
    return Promise.all([
        new Promise<void>(resolve => setTimeout(() => resolve(), t)),
        advance(t)
    ])
}

async function impl(
    log: string[],
    waitImpl: (t: number) => Promise<unknown> | unknown
) {
    // event handler
    void (async () => {
        log.push('event')
        await wait(0)
        log.push('then')
        void wait(20).then(() => {
            log.push('after 20')
        })
        await wait(0)
        log.push('next')
        await wait(0)
        log.push('x')
        await wait(0)
        log.push('y')
        await wait(0)
        log.push('z')
        await wait(100)
        log.push('delayed')
    })()

    // delay
    await waitImpl(50)
    log.push('continue')
}

describe.each([false, true])('usePromise: %s', (useAwaitAdvance) => {
    test.each([false, true])('fakeTimers: %s', async (useFakeTimers) => {
        const log: string[] = []
    
        if (useFakeTimers) {
            jest.useFakeTimers()
        }
    
        await impl(
            log,
            useAwaitAdvance
                ? (t) => waitAwaitAdvance(
                    t,
                    useFakeTimers
                        ? async (u: number) => {
                            do {
                                jest.advanceTimersByTime(1)
                                for(let before;;) {
                                    before = jest.getTimerCount()
                                    await new Promise<void>(r => r())
                                    if (jest.getTimerCount() === before) {
                                        break
                                    }
                                    jest.advanceTimersByTime(0)
                                }
                            } while(--u > 0)
                        }
                        : () => Promise.resolve()
                )
                : (t) => waitVoidAdvance(
                    t,
                    useFakeTimers
                        ? jest.advanceTimersByTime
                        : () => undefined
                ),
        )
 
        jest.useRealTimers()
        await wait(10)

        expect(log).toEqual([
            'event',
            'then',
            'next',
            'x',
            'y',
            'z',
            'after 20',
            'continue',
        ])
    })

})
```

Reply:
The code above just demonstrates the problem in a reproducible manner.
In real-world scenarios this usually isn't that obvious, the pushes to the event loop can be implemented in nested/imported modules and/or can involve event-driven APIs.

> ```ts
> (delay: number) => {
>    (async () => {
>        /* do your crazy async work */
>    })();
>}
>```
^ This would not allow the asynchronous code after any `await` to run before we continue with the next interaction that is supposed to happen "later".
The problem is that fake timers create a situation that normally doesn't exist: Microtasks being queued and not executed before picking the next "macrotask". This is because the next "macrotask" is actually queued synchronously while the microtask still is queued through the event loop.

Reply:
Got it. I don't understand why anyone would ever do this. But sure, I can make it support promises too if someone wants to do this.